test.js

```diff
@@ -1,5 +1,8 @@
 /**
- * @typedef {import('unist').Node} Node
+ * @typedef {{type: 'leaf', value: string}} Leaf
+ * @typedef {{type: 'node', children: (Node | Leaf)[]}} Node
+ * @typedef {{type: 'root', children: (Node | Leaf)[]}} Root
+ * @typedef {Root | Node | Leaf} AnyNode
```

Comment:
```suggestion
 * @typedef {{type: 'node', children: Array<Node | Leaf>}} Node
 * @typedef {{type: 'root', children: Array<Node | Leaf>}} Root
 * @typedef {Root | Node | Leaf} AnyNode
```

I prefer `Array<>` syntax.